File: Backend/Entities/TransferredCourseGroup.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Entities
{
    // Group of transferred courses

    /// <summary>A class for representing a group of courses that have been transferred from another institution..</summary>
    public class TransferredCourseGroup
    {
        //[ForeignKey("CTEForm")]
        //public Guid CTEFormId { get; set; }
        [Key]
        public Guid Id { get; set; }

        [Required]
        public ICollection<TransferredCourse> TransferredCourses { get; set; }
        [Required]
        public ExemptedCourse ExemptedCourse { get; set; }
    }
}

File: Frontend/src/app/_models/transferred-course-group.ts
import { GUID } from '../../utils/guid';
import { TransferredCourse } from './transferred-course';
import { ExemptedCourse } from './exempted-course';

export interface TransferredCourseGroup {
  id: GUID;
  transferredCourses: TransferredCourse[];
  exemptedCourse: ExemptedCourse;
}
